src/main/java/pl/fracz/mcr/source/SourceFile.java

```diff
@@ -0,0 +1,135 @@
+package pl.fracz.mcr.source;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.view.View;
+import pl.fracz.mcr.comment.CommentNotAddedException;
+import pl.fracz.mcr.preferences.ApplicationSettings;
+import pl.fracz.mcr.syntax.PrettifyHighlighter;
+import pl.fracz.mcr.syntax.SyntaxHighlighter;
+import pl.fracz.mcr.util.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.StringTokenizer;
+
+public class SourceFile {
+    private static final SyntaxHighlighter SYNTAX_HIGHLIGHTER = new PrettifyHighlighter();
```

Comment:
We're stuck with PrettifyHighlighter, we could use dependency injection for flexibility